lambdarest/__init__.py

```diff
@@ -106,18 +107,24 @@ def inner_lambda_handler(event, context=None):
         path = event["path"].lower()
         method_name = event["httpMethod"].lower()
         func = None
+        kwargs = {}
         error_tuple = ("Internal server error", 500)
         logging_message = "[%s][{status_code}]: {message}" % method_name
         try:
-            func = http_methods.get(path, http_methods.get("*", {}))[method_name]
+            mapping = url_maps.bind('example.com', '')
+            func, kwargs = mapping.match(path, method=method_name)
+        except NotFound as e:
+            logging.warning(logging_message.format(
+                status_code=404, message=str(e)))
+            error_tuple = (str(e), 404)
         except KeyError:
             logging.warning(logging_message.format(
                 status_code=405, message="Not supported"))
             error_tuple = ("Not supported", 405)
```

Comment:
What raises the KeyError and tells us that the HTTP method is unsupported?

Reply:
The KeyError was from before, removed it, not needed anymore.